### PR TITLE
Added zod validation for id in GET topic/:id route

### PR DESCRIPTION
**Pull request by sparsh-18**
Below is the attached screenshot for 3 cases,
image

image

image

### CODE REVIEW

File: packages/server/src/api/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Router } from 'express';
 import { PrismaClientSingleton } from '../utils';
 import { emailValidator, idValidator, userTopicsValidator } from '../validators';
-import { ZodError } from 'zod';
+import { ZodError, z } from 'zod';
```
Comment (prasenjeet-symon) [inline packages/server/src/api/index.ts:4]: You did a great job but can you please add your ZOD validation to the folder `validators`. See the other validations

Comment (prasenjeet-symon) [reply packages/server/src/api/index.ts:4]: Import it then use it also try to extend from the existing Zod validation. Learn from the old validation logic

Comment (sparsh-18) [reply packages/server/src/api/index.ts:4]: sure, thanks for the input

Comment (sparsh-18) [reply packages/server/src/api/index.ts:4]: Hi @prasenjeet-symon
Should I refine the existing idValidator with the "not 0" condition or should I add a new paramIdValidator?
Which one would you prefer?

Comment (prasenjeet-symon) [reply packages/server/src/api/index.ts:4]: Modify the existing one since any MYSQL id cannot be 0.

Comment (sandheep45) [reply packages/server/src/api/index.ts:4]: Refining existing validator would we a better choice

**Comment by sparsh-18** [comment]
Did the changes, please check.

File: packages/server/src/api/index.ts [hunk 2]
```diff
@@ -171,25 +171,16 @@ router.get('/user/topics', async (req, res) => {
...
-    const idSchema = z
-        .number({
-            required_error: 'id is required',
-            invalid_type_error: 'id must be a number'
-        })
-        .refine((value) => value !== 0, {
-            message: 'id must not be equal to 0',
-        });
-
+    let parsedParam;
```
Comment (prasenjeet-symon) [inline packages/server/src/api/index.ts:174]: Avoid usage of `let` keyword with unassigned variable.

Comment (prasenjeet-symon) [reply packages/server/src/api/index.ts:174]: Hope you understand why i am reviewing your code so critically. This is just to make sure this project is at it's best. Thanks

File: packages/server/src/validators/index.ts
Comment (prasenjeet-symon) [inline packages/server/src/validators/index.ts:1]: Everything is good on this file. Great job 👍

**Comment by sandheep45** [comment]
@prasenjeet-symon i would like to add one point here. Moving the prisma calls along with the validator parsing into try block would be a better approach. It will catch any error (zod error or prisma error) in the catch block thus will send 400 error to the fe. This way will also eliminate the use of `let` keyword.

**Comment by prasenjeet-symon** [comment]
> @prasenjeet-symon i would like to add one point here. Moving the prisma calls along with the validator parsing into try block would be a better approach. It will catch any error (zod error or prisma error) in the catch block thus will send 400 error to the fe. This way will also eliminate the use of `let` keyword.

@sparsh-18 Can you submit the new PR

**Review by prasenjeet-symon — CHANGES_REQUESTED** [review]
Waiting for new changes

File: packages/server/src/api/index.ts (hunk 2 above)
Comment (sparsh-18) [reply packages/server/src/api/index.ts:174]: Yes I understand and I really like your fast and on point feedback.

**Comment by sparsh-18** [comment]
> Waiting for new changes

sure

**Comment by sparsh-18** [comment]
Hi, added the changed

**Review by prasenjeet-symon — APPROVED** [review]
Awesome 👍

**Pull request merged**